### ISSUE TITLE
Can it be implement with EditText ?

### ISSUE THREAD

**Issue by mdevspot** [noties/Markwon#97]
I was trying to implement it with EditText,  when setting
 
    override fun afterTextChanged(s: Editable?) {
            myEditText.removeTextChangedListener(this) //Removing listener to prevent infinite loop
            Markwon.setMarkdown(myEditText,s.toString())
            myEditText.addTextChangedListener(this) //Adding listener for text change
     }

the problems are below...
1> Whitespaces are trimmed.
2> Not able to set cursor position manually because of Prob@1
3> String length is not counting markdowns
4> Not able to remove markdown, Needed to remove whole word or sentence with markwdown

**Comment by noties** [comment]
Hello @mdevspot !

First of all, a minor change to code snippet that you'd shared:
```kotlin
override fun afterTextChanged(s: Editable) {
        // there is no need to add/remove listener here, this method will be 
        // continuously called by EditText when text changes
        Markwon.setMarkdown(myEditText, s.toString())
 }
```

But anyway, do not do it, because...

Unfortunately `Markwon` is not suitable for editable markdown. It _might be_ but there is no support for it now. Here is why:

* As you had mentioned, source positions are not persisted/tracked. It is due to `commonmark-java` implementation (it's a good thing actually as it comes with great performance). 
* Each call to `Markwon#setMarkdown` will replace the whole text inside `EditText`. This is why cursor position is not persisted (did you know that cursor-position/text-selection represented as a `Span`?).

Basically, you most likely want a markdown highlight functionality when editing. This way all the markdown _syntax_ will be preserved. But, `Markwon` has no such feature. You can use `Markwon` to display markdown, but not edit.

There is a project (that is a fork of `commonmark-java`) -> [flexmark-java](https://github.com/vsch/flexmark-java). It was developed especially for editing (there is a Intellij plugin for markdown editing based on `flexmark-java` by its author). It has source positions tracked. And also has bigger set of supported markdown features. Maybe you can use it for your purposes.
